src/autopas/containers/verletClusterLists/VerletClusterLists.h

```diff
@@ -348,6 +383,69 @@ class VerletClusterLists : public ParticleContainer<FullParticleCell<Particle>>
     }
   }
 
+  /**
+   * Calculates a cluster thread partition that aims to give each thread about the same amount of cluster pair
+   * interactions, if each thread handles the neigbhors of all clusters it gets assigned.
+   */
+  void calculateClusterThreadPartition() {
+    size_t numClusterPairs = 0;
+    this->template traverseClusters<false>(
+        [&numClusterPairs](auto &cluster) { numClusterPairs += cluster.getNeighbors().size(); });
+
+    constexpr int minNumClusterPairsPerThread = 1000;
+    auto numThreads =
+        std::clamp(static_cast<int>(numClusterPairs / minNumClusterPairsPerThread), 1, autopas_get_max_threads());
+
+    size_t numClusterPairsPerThread = numClusterPairs / numThreads;
+    fillClusterRanges(numClusterPairsPerThread, numThreads);
+  }
+
+  /**
+   * Fills in the cluster ranges of the cluster thread partition.
```

Comment:
Extend this description a bit. What does it actually do?
sth. like "gives each thread an appropriate range of clusters, s.t., the number of cluster pairs is similar."